samples_project/Assets/SampleViewer/Samples/FeatureLayer/FeatureLayer.cs

```diff
@@ -0,0 +1,407 @@
+using Esri.ArcGISMapsSDK.Components;
+using Esri.ArcGISMapsSDK.Utils.GeoCoord;
+using Esri.GameEngine.Geometry;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using TMPro;
+using UnityEngine.Networking;
+using UnityEngine;
+using UnityEngine.UI;
+
+[System.Serializable]
+public struct WebLink
+{
+    public string Link;
+    public string[] RequestHeaders;
+}
+
+[System.Serializable]
+public class Feature
+{
+    public Geometry geometry = new Geometry();
+    public Properties properties = new Properties();
+}
+
+[System.Serializable]
+public class Properties
+{
+    public List<string> propertyNames = new List<string>();
+    public List<string> data = new List<string>();
+}
+
+[System.Serializable]
+public class Geometry
+{
+    public List<double> coordinates = new List<double>();
+}
+
+public class FeatureLayer : MonoBehaviour
+{
+    [SerializeField] private ArcGISCameraComponent arcGISCamera;
+    [SerializeField] private Transform contentContainer;
+    [SerializeField] private GameObject featurePrefab;
+    private int featureSRWKID = 4326;
+    [SerializeField] private List<string> outfields = new List<string>();
+    private int stadiumSpawnHeight = 10000;
+    private FeatureLayerUIManager UIManager;
+
+    public List<Feature> Features = new List<Feature>();
+    public bool GetAllFeatures = true;
+    public bool GetAllOutfields = true;
+    public JToken[] jFeatures;
+    public int LastValue = 1;
+    public List<GameObject> FeatureItems = new List<GameObject>();
+    public List<Toggle> ListItems = new List<Toggle>();
+    public bool NewLink = true;
+    public GameObject OutfieldItem;
+    public List<string> OutfieldsToGet = new List<string>();
+    public int StartValue;
+    public WebLink WebLink;
+
+    private void Start()
+    {
+        CreateLink(WebLink.Link);
+        StartCoroutine(GetFeatures());
+        UIManager = GetComponent<FeatureLayerUIManager>();
+    }
+
+    public void CreateLink(string link)
+    {
+        EmptyOutfieldsDropdown();
+        if (link != null)
```

Comment:
Would this ever be null here?

Reply:
likely no, but it is possible. Part of the feedback from Unreal was that users should be able to clear the input field all together if that want. So if they do that it would be null and we don't want to create the link in that case.